findAndReplacePassportValue/helpers.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Find a current  passport variable (fn) or (val) and replace it in each node in a flow (live or published
+ */
+
+const updateNodeFn = (flowData, currentPassportVariable, newPassportVariable) => {
+  let newFlowData = flowData;
+  Object.entries(flowData)
+    .filter(([_nodeId, nodeData]) => nodeData?.["data"]?.["fn"] || nodeData?.["data"]?.["val"])
+    .forEach(([nodeId, nodeData]) => {
+      const passportKey = nodeData["data"]["fn"] ? "fn" : "val"
+      const currentFn = nodeData["data"][`${passportKey}`]
+      const currentValueSegments = currentFn.split(".")
+      const matchIndexPosition = currentValueSegments.indexOf(currentPassportVariable)
+      if (matchIndexPosition !==-1) {
```

Comment:
If we look at real ProjectTypes, we can see that there are edge cases where the same word can appear twice - eg https://github.com/theopensystemslab/digital-planning-data-schemas/blob/main/types/enums/ProjectTypes.ts#L98

Is it actually necessary to `split(".")` the fn/val string here and then re-join, can we simply try a `replaceAll()` on the full passport string - which would then also give us more flexibility if we want to use this script ot replace _full_ passport variables (any number characters/segments) rather than only a single segment of them?

Reply:
Thanks for sharing the `projectTypes` and checking the assumption I'd made!

I'll have another look at this functionality. I had been using `.replace` the reason I was splitting was so I could do exact matches as I was worried about incorrectly replacing substrings so for something like `alter → change` incorrectly leadings to`alteration.openings → changeation.openings`

I'll have another look at `replaceAll` to see how whole sections can be updated 👍

Reply:
Fixed here: 4bd01282d2359b91010b784c76374f7d879f32ff

Reply:
Oops sorry left this as a new comment rather than on this thread: https://github.com/theopensystemslab/planx-data-migrations/pull/9#discussion_r1494585500